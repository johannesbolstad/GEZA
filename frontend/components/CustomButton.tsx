import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { Button } from 'react-native-paper';
import { IconSource } from 'react-native-paper/lib/typescript/components/Icon';

type ButtonProps = {
    icon: IconSource;
    onPress: () => void;
    text: string;
    grey?: boolean;
};

/**
 * This is a custom button that takes four props
 * @param icon is the icon the button has
 * @param onPress is a function that gets called on a press event.
 * @param onPress is the text on the button
 * @param grey is a boolean that decides whether the button should be grey ot not.
 */

const CustomButton: React.FC<ButtonProps> = ({ icon, onPress, text, grey }) => {
    return (
        <View style={styles.subcontainer}>
            <Button
                testID={'myButton'}
                icon={icon}
                labelStyle={{ color: 'white' }}
                style={styles.button}
                mode="contained"
                color={grey ? '#BEBEBE' : '#4E8E48'}
                onPress={onPress}
                accessible={true}
                accessibilityLabel={text}
            >
                <Text style={{ color: 'white' }}> {text} </Text>
            </Button>
        </View>
    );
};
const styles = StyleSheet.create({
    subcontainer: {
        flexDirection: 'row'
    },
    button: {
        marginTop: 20,
        marginRight: 20,
        borderRadius: 5,
        alignItems: 'center',
        color: '#87B18A',
        borderWidth: 3,
        borderColor: 'black'
    }
});
export default CustomButton;
