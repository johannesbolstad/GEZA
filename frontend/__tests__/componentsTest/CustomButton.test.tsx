import React from 'react';
import renderer, { act, ReactTestRendererJSON } from 'react-test-renderer';
import CustomButton from '../../components/CustomButton';

describe('MessageComponent', () => {
    const onSubmit = jest.fn();
    const tree = renderer
        .create(
            <CustomButton icon={'check'} text={'Sign Up'} onPress={onSubmit} />
        )
        .toJSON() as ReactTestRendererJSON[];
    const tree2 = renderer.create(
        <CustomButton icon={'check'} text={'Sign Up'} onPress={onSubmit} />
    );

    it('Renders correctly', () => {
        expect(tree).toMatchSnapshot();
    });

    it('should fire onSubmit events', () => {
        const button = tree2.root.findByProps({ testID: 'myButton' }).props;
        act(() => button.onPress());
        expect(onSubmit).toHaveBeenCalled();
    });
});
