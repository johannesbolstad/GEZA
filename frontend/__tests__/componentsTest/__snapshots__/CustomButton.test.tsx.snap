// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageComponent Renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#4E8E48",
        "borderColor": "black",
        "borderRadius": 5,
        "borderStyle": "solid",
        "borderWidth": 3,
        "color": "#87B18A",
        "elevation": 2,
        "marginRight": 20,
        "marginTop": 20,
        "minWidth": 64,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.75,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 1.5,
      }
    }
  >
    <View
      accessibilityLabel="Sign Up"
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": undefined,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 5,
          },
        ]
      }
      testID="myButton"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "marginLeft": 12,
              "marginRight": -4,
            }
          }
        >
          <Text />
        </View>
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "letterSpacing": 1,
                  "marginHorizontal": 16,
                  "marginVertical": 9,
                  "textAlign": "center",
                },
                undefined,
                Object {
                  "textTransform": "uppercase",
                },
                Object {
                  "color": "#ffffff",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "color": "white",
                },
              ],
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "white",
              }
            }
          >
             
            Sign Up
             
          </Text>
        </Text>
      </View>
    </View>
  </View>
</View>
`;
